xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/TaskManager.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.index;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.index.internal.TaskData;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Provide the operations to interact with the task manager.
+ *
+ * @version $Id$
+ * @since 14.1RC1
+ */
+@Role
+@Unstable
+public interface TaskManager
+{
+    /**
+     * Add a task to the queue.
+     *
+     * @param wikiId the wiki containing the document
+     * @param docId the document id
+     * @param version the document version
+     * @param kind the kind of task to add
+     * @return a completable future for this task
+     */
+    CompletableFuture<TaskData> addTask(String wikiId, long docId, String version, String kind);
+
+    /**
+     * Replace all the tasks of the queue with the same document and task kind with the new task.
+     *
+     * @param wikiId the wiki containing the document
+     * @param docId the document id
+     * @param version the document version
+     * @param kind the kind of task to add
+     * @return a completable future for this task
+     */
+    CompletableFuture<TaskData> replaceTask(String wikiId, long docId, String version, String kind);
+
+    /**
+     * Starts the consumer thread.
+     */
+    void startThread();
```

Comment:
Is this public API really needed ? I tend to do this kind of thing in the `initialize()` of the component.

Reply:
I agree but if I remove this method from the API, I have to make a cast in `TaskApplicationReadyListener`.

Reply:
I don't understand the relationship. For me if you initialize the thread in DefaultTaskManager#initialize() then TaskApplicationReadyListener becomes totally useless (since it's only point seems to be to start the thread).